service/controller/resource/masters/create_delete_legacy_vmss.go

```diff
@@ -0,0 +1,42 @@
+package masters
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/azure-operator/service/controller/internal/state"
+	"github.com/giantswarm/azure-operator/service/controller/key"
+)
+
+func (r *Resource) deleteLegacyVMSSTransition(ctx context.Context, obj interface{}, currentState state.State) (state.State, error) {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return Empty, microerror.Mask(err)
+	}
+
+	// Delete the scaleset
+	err = r.deleteScaleSet(ctx, key.ResourceGroupName(cr), key.LegacyMasterVMSSName(cr))
+	if IsScaleSetNotFound(err) {
+		// Scale set not found, all good.
+		return DeploymentCompleted, nil
```

Comment:
Is there any chance that scale set deletion went through but somehow still returned an error and on next reconciliation round we skip `UnblockAPICalls` even though it would need to be done?

Reply:
Maybe there is, but I never witnessed it.
I am not sure how to handle this differently.